docs/get-started/platforms/stm32.md

```diff
@@ -5,13 +5,111 @@ LVGL Can be added to [STM32CubeIDE](https://www.st.com/en/development-tools/stm3
 
 Now that the source files are included in your project, follow the instructions for [Porting](https://docs.lvgl.io/master/porting/project.html) your project to create the ```lv_conf.h``` file, and initialise the display.
 
-## Example
+## Bare Metal Example
+A minimal example using STM32CubeIDE, and HAL.
+* When setting up **Pinout and Configuration** using the **Device Configuration Tool**, select **System Core** - **SYS** - **Timebase Source** and set to **TIM1** or any timer of your choice (anything other than SysTick).
+* Configure any other peripherals (including the LCD panel), and initialise them in *main.c*.
+* ```#include "lvgl.h"``` in the *main.c* file.
+* Create some frame buffer(s) as global variables:
+```
+//Frame buffers
+/*A static or global variable to store the buffers*/
+static lv_disp_draw_buf_t disp_buf;
+
+/*Static or global buffer(s). The second buffer is optional*/
+static lv_color_t buf_1[BUFF_SIZE]; //TODO: Chose a buffer size. DISPLAY_WIDTH * 10 is one suggestion.
+static lv_color_t buf_2[BUFF_SIZE];
+```
+* In your ```main()``` function, after initialising your CPU, peripherals, and LCD panel, call ```lv_init();``` to initialise LVGL. You can then register the frame buffers using ```lv_disp_draw_buf_init()```, and create the display driver using ```lv_disp_drv_init()```.
+```
+//Initialise LVGL UI library
+lv_init();
+lv_disp_draw_buf_init(&disp_buf, buf_1, NULL, BUFF_SIZE);
+
+static lv_disp_drv_t disp_drv;          /*A variable to hold the drivers. Must be static or global.*/
+lv_disp_drv_init(&disp_drv);            /*Basic initialization*/
+disp_drv.draw_buf = &disp_buf;          /*Set an initialized buffer*/
+disp_drv.flush_cb = my_flush_cb;        /*Set a flush callback to draw to the display*/
+disp_drv.hor_res = WIDTH;                 /*Set the horizontal resolution in pixels*/
+disp_drv.ver_res = HEIGHT;                 /*Set the vertical resolution in pixels*/
+
+lv_disp_t * disp;
+disp = lv_disp_drv_register(&disp_drv); /*Register the driver and save the created display objects*/
+```
+* Create some dummy objects to test the output:
+```
+// Change the active screen's background color
+lv_obj_set_style_bg_color(lv_scr_act(), lv_color_hex(0x003a57), LV_PART_MAIN);
+lv_obj_set_style_text_color(lv_scr_act(), lv_color_hex(0xffffff), LV_PART_MAIN);
+
+/*Create a spinner*/
+lv_obj_t * spinner = lv_spinner_create(lv_scr_act(), 1000, 60);
+lv_obj_set_size(spinner, 64, 64);
+lv_obj_align(spinner, LV_ALIGN_BOTTOM_MID, 0, 0);
+```
+* Add a call to ```lv_timer_handler()``` inside your ```while(1)``` loop:
+```
+/* Infinite loop */
+while (1)
+{
+  lv_timer_handler();
+  HAL_Delay(5);
+}
+```
+* Add a call to ```lv_tick_inc()``` inside the ```HAL_TIM_PeriodElapsedCallback()``` function, near the bottom of *main.c*, which should have been generated after saving your *Device Configuration* in the first step.
```

Comment:
I usually just do this:
```c
void SysTick_Handler(void)
{
	HAL_IncTick();
	HAL_SYSTICK_IRQHandler();
	lv_tick_inc(1);
#ifdef USE_RTOS_SYSTICK
	osSystickHandler();
#endif
}
```

I think it's the simplest option. Or are there any issues with it?

Reply:
Hi,
Thanks for checking my submission. New to LVGL but I'm enjoying it and happy to contribute. 
I was also trying to decide on using ```SysTick_Handler```, or the timer interrupt method and I just already had timers in front of me from the FreeRTOS example. I think your example is more simple, thanks.